src/modules/job-ingest/validate.c

```diff
@@ -89,15 +99,29 @@ struct validate *validate_create (flux_t *h,
 
     if (executable_is_intree ())
         conf_flags |= CONF_FLAG_INTREE;
-    argv[0] = (char *)validate_path,
-    argv[1] = "--schema";
-    argv[2] = (char *)schema_path,
-    argv[3] = NULL;
-    assert (argv[0] != NULL);
-    assert (argv[2] != NULL);
+
+    if (str_ends_with (validate_path, ".py")) {
+        argv[0] = PYTHON_INTERPRETER;
+        argv[1] = (char *)validate_path,
+        argv[2] = "--schema";
+        argv[3] = (char *)schema_path,
+        argv[4] = NULL;
+        argc = 4;
+        assert (argv[1] != NULL);
+        assert (argv[3] != NULL);
+    } else {
+        argv[0] = (char *)validate_path,
+        argv[1] = "--schema";
+        argv[2] = (char *)schema_path,
+        argv[3] = NULL;
+        argv[4] = NULL;
+        argc = 3;
+        assert (argv[0] != NULL);
+        assert (argv[2] != NULL);
+    }
```

Comment:
Suggestion to reduce code duplication (untested):
```c
assert (validate_path != NULL);
assert (schema_path != NULL);
argc=0;
if (str_ends_with (validate_path, ".py"))
    argv[argc++] = PYTHON_INTERPRETER;
argv[argc++] = (char *)validate_path;
argv[argc++] = "--schema";
argv[argc++] = (char *)schema_path,
argv[argc++] = NULL;
```

Reply:
Nice! The duplicated code bothered me as well, but I didn't think of `argc++`.  Done!